iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergSerDe.java

```diff
@@ -105,8 +106,18 @@ public void initialize(@Nullable Configuration configuration, Properties serDePr
     } else {
       try {
         Table table = IcebergTableUtil.getTable(configuration, serDeProperties);
-        // always prefer the original table schema if there is one
-        this.tableSchema = table.schema();
+        if (serDeProperties.containsKey(serdeConstants.AS_OF_VERSION)) {
+          this.tableSchema = SnapshotUtil.schemaFor(table,
+              Long.valueOf(serDeProperties.getProperty(serdeConstants.AS_OF_VERSION)));
+          serDeProperties.remove(serdeConstants.AS_OF_VERSION);
```

Comment:
Let's explain in a comment why we need to remove this from serdeProperties, and perhaps also why we need serdeProperties to carry this information nstead of job configuration in the first place.